### PR TITLE
Add error handling to creating and removing lowercase views

### PR DESCRIPTION
**Pull request by Will-Howard**
We had an error on a prod deployment where there was a table (which I created for some admin reason) which already had a lowercase name, and `remove_lowercase_views()` tried to run `drop view ...` on it. This caused an error because it was a table and not a view.

This PR adds error handling which:
 - checks for this specific case where a table name is already in lowercase
 - adds generic error handling to every attempt to `drop view`, because we don't want this to cause deployments to fail

### CODE REVIEW

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    DECLARE
+        rec record;
+    BEGIN
+      FOR rec IN
+        select table_name,
+          string_agg(format('%I as %s', column_name, column_name), ', ') cols
+        from information_schema."columns"
+        inner join information_schema."tables" using (table_name)
+        where "columns".table_schema = 'public'
+          and table_name not in ('migration_log', 'mongo2pg_lock')
```
Comment (Will-Howard) [inline packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:25]: I'm not sure I fully understand this part, but the important thing is that in the loop below (and the equivalent loop in `remove_lowercase_views`) rec.table_name goes through the _mixed case_ name of every table in the public schema

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  action: async () => {
+    const db = getSqlClientOrThrow();
+
+    const sql = `
+    create or replace function refresh_lowercase_views() returns void as
+    $$
+    DECLARE
+        rec record;
+    BEGIN
+      FOR rec IN
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:19]: I would document what shape rec will take

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        select table_name,
+          string_agg(format('%I as %s', column_name, column_name), ', ') cols
+        from information_schema."columns"
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:22]:
```suggestion
        from information_schema.columns
```

I don't think you need any of these quotes in this function.

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts (hunk 1 above)
Comment (jpaddison3) [reply packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:25]:
```
Get all tables. Join against all columns in the db and group by tables.
```

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts (hunk 2 above)
Comment (jpaddison3) [reply packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:19]:
```
table_name | AdvisorRequests
cols       | "jobAds" as jobAds, "schemaVersion" as schemaVersion, "createdAt" as createdAt, "legacyData" as legacyData, _id as _id, "userId" as userId, "interestedInMetaculus" as intereste
dInMetaculus
```

File: packages/lesswrong/server/manualMigrations/index.ts [hunk 4]
```diff
@@ -120,3 +120,4 @@ import './2023-02-10-fillCreatedAtForDebouncerEvents'
 import './2023-03-16-createUserActivities'
 import './2023-04-10-importACXmeetups'
 import './2023-06-13-renameShortformToQuicktakes'
+import './2023-07-28-viewUpdaterErrorHandling'
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/index.ts:123]: Can you add a comment to `packages/lesswrong/server/manualMigrations/2023-01-13-createViewUpdater.ts` saying "not the most recent version" or similar?

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts (hunk 3 above)
Comment (jpaddison3) [reply packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:22]: I acknowledge that you haven't changed the original function here.

**Review by jpaddison3 — APPROVED** [review]

File: packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts (hunk 3 above)
Comment (Will-Howard) [reply packages/lesswrong/server/manualMigrations/2023-07-28-viewUpdaterErrorHandling.ts:22]: I agree they're not necessary. I'd rather keep them though seeing as were doing a lot of case-fiddling here, I think it makes it less confusing to have everything in quotes by default so you can more easily work out in which parts the case is actually important

**Comment by Will-Howard** [comment]
I ran this on staging and prod just now (around 2023-07-31T11:14:58.708Z)

**Pull request merged**